src/Subscriptions/StorageManager.php

```diff
@@ -118,19 +118,42 @@ public function storeSubscriber(Subscriber $subscriber, string $topic): void
     /**
      * Delete subscriber.
      *
+     * @param  string  $channel
      * @return \Nuwave\Lighthouse\Subscriptions\Subscriber|null
      */
     public function deleteSubscriber(string $channel): ?Subscriber
     {
         $key = self::SUBSCRIBER_KEY.".{$channel}";
         $hasSubscriber = $this->cache->has($key);
-
         $subscriber = $this->cache->get($key);
 
         if ($hasSubscriber) {
+            $this->removeSubscriberFromTopic($subscriber, $key);
             $this->cache->forget($key);
         }
 
         return $subscriber;
     }
+    
+    /**
+     * Remove the subscriber from the topic they are subscribed to.
+     *
+     * @param  Subscriber $subscriber
+     * @param  string $subscriberKey
+     */
+    public function removeSubscriberFromTopic(Subscriber $subscriber, string $subscriberKey)
+    {
+        $key = strtoupper(Str::snake($subscriber->operationName));
+        $topicKey = self::TOPIC_KEY.".{$key}";
+
+        $topic = Collection::make(
+            $this->cache->has($topicKey)
+                ? json_decode($this->cache->get($topicKey), true)
+                : []
+        )->filter(function ($key) use ($subscriberKey) {
+            return $key === $subscriberKey;
+        });
```

Comment:
```suggestion
        )->reject(function ($key) use ($subscriberKey) {
            return $key === $subscriberKey;
        });
```

Otherwise the only thing that remains is the one you wanted to remove :)

Reply:
👍🏾